lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -47,11 +48,41 @@ sub _git_clone_all ($job, $clones) {
 
     # iterate clones sorted by path length to ensure that a needle dir is always cloned after the corresponding casedir
     for my $path (sort { length($a) <=> length($b) || $a cmp $b } keys %$clones) {
-        my $url = $clones->{$path};
+        my $url = $clones->{$path} // '';
         die "Don't even think about putting '..' into '$path'." if $path =~ /\.\./;
+
+        my $git = OpenQA::Git->new(app => $app, dir => $path);
+        $url //= $git->get_origin_url;
+        my $server_host;
+        if ($url =~ m{^(?:\w+://)?([^/\@:\s]+)(?::\d+)?/}) {
```

Comment:
to me, looks like you dont need the `(?::\d+)?/`. or is there any particular reason?

Reply:
Yeah not sure it was really needed here but it was meant to optionally match a port number. Now this regex is replaced by `Mojo::URL->host` though